### PR TITLE
feat: add form config for reading certs for PG

### LINKED ISSUE #31400: Form config changes for SSL configs of MTLS
Sub task of the issue #31326

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Form config changes for the PG datasource to support reading certs from user

Fixes #31400 

## Automation

/ok-to-test tags="tag.Datasource"

### :mag: Cypress test results

> [!IMPORTANT]  
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8538007537>
> Commit: `0fb57fe78e973db8a3df5f2e407b85eef4adb4e2`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8538007537&attempt=1" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

## Summary by CodeRabbit

- **New Features**
	- Enhanced SSL configuration options for database connections, including support for client and server CA certificate files.
- **Refactor**
	- Improved handling and encoding of SSL certificate content for database connections, ensuring compatibility and security.

### CODE REVIEW

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json [hunk 1]
```diff
@@ -103,6 +103,135 @@
...
+                    },
+                    {
+                      "label": "Base64 String",
+                      "value": "BASE64_STRING"
+                    }
+                  ],
+                  "hidden": {
+                    "path": "datasourceConfiguration.connection.ssl.authType",
+                    "comparison": "NOT_EQUALS",
+                    "value": "ENABLED"
```
Comment (AmanAgarwal041) [inline app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json:146]: This should be `VERIFY_CA`. As there is not `ENABLED` value for key `datasourceConfiguration.connection.ssl.authType`.

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json [hunk 2]
```diff
@@ -103,6 +103,135 @@
...
```
Comment (AmanAgarwal041) [inline app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json:202]: This should be `VERIFY_CA`. As there is not `ENABLED` value for key `datasourceConfiguration.connection.ssl.authType`.

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json [hunk 3]
```diff
@@ -103,6 +103,135 @@
...
+                  "encrypted": true,
+                  "hidden": {
+                    "path": "datasourceConfiguration.connection.ssl.caServerCertificateType",
+                    "comparison": "NOT_EQUALS",
+                    "value": "BASE64_STRING"
+                  }
+                }
+              ]
+            },
+            {
```
Comment (AmanAgarwal041) [inline app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json:232]: This empty object occupies the space. Please remove this if not needed.

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json (hunk 1 above)
Comment (AmanAgarwal041) [reply app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json]: Done

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json (hunk 3 above)
Comment (AmanAgarwal041) [reply app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json]: Done

File: app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json (hunk 2 above)
Comment (AmanAgarwal041) [reply app/server/appsmith-plugins/postgresPlugin/src/main/resources/form.json]: Done

**Review by sumitsum — APPROVED** [review]

**Pull request merged**